### DISCUSSION TITLE
using ref and useState in composables

### DISCUSSION THREAD

**Discussion by azoom-nguyen-van-nam** [nuxt/nuxt#15972] [Questions]
In Doc Nuxt 3:
https://nuxt.com/docs/getting-started/state-management#best-practices

Can someone help me explain this part?
When will ref in composables be used, and when will useState be used?

**Comment by fathi91** [comment]
you should always use `useState` outside of `<script setup>` 
and insaid `<script setup>`  you would use `useState` if you have any Shared State that you want to access by key other then that use `ref`
